### PR TITLE
DG update for recurrence backlogging

### PR DESCRIPTION
**Pull request by NigelYeoTW**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -164,79 +170,133 @@ recurrence it is (e.g. `DailyRecurrence`, `null`) which is encapsulated as a `Re
...
 for brevity.
 
 Functionality: <br>
 1. Checks `lastRecurred` attribute of `recurringIncome`/`recurringSpending` against the current date via `LocalDate.now`
 2. According to the type of recurrence, check if enough time has passed between the 2 dates
 3. Adds additional recurring entries into the `IncomeList`/`SpendingList` if needed.
 
+Note: <br>
+Parameter `isAdding` is illustrated in the sequence diagram above in the opt condition. It allows the method to add 
+entries to list if set to `true`, it may be set to false when used for backlogging
```
Comment (wx-03) [inline docs/DeveloperGuide.md:213]: maybe the note can be moved to inside the sequence diagram instead

Comment (NigelYeoTW) [reply docs/DeveloperGuide.md:213]: alrighty

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -164,79 +170,133 @@ recurrence it is (e.g. `DailyRecurrence`, `null`) which is encapsulated as a `Re
...
-public void checkIncomeRecurrence(Income recurringIncome, IncomeList incomes)
+public void checkIncomeRecurrence(Income recurringIncome, IncomeList incomes, boolean isAdding)
 @Override
-public void checkSpendingRecurrence(Spending recurringSpending, SpendingList spendings)
+public void checkSpendingRecurrence(Spending recurringSpending, SpendingList spendings, boolean isAdding)
 ```
 Below illustrates the functionality of the checkIncomeRecurrence method through a sequence diagram <br>
 <br>
-<img src="./Diagrams/Recurrence/addRecurrenceEntry.png" alt="addRecurrenceEntry" width="800"/> <br>
+<img src="./Diagrams/Recurrence/addRecurrenceEntrySD.png" alt="addRecurrenceEntrySD" width="800"/> <br>
```
[the remark this answers is not in the document]
Comment (NigelYeoTW) [reply docs/DeveloperGuide.md:201]: roger roger, thanks

[the remark this answers is not in the document]
Comment (NigelYeoTW) [reply docs/DeveloperGuide.md:201]: can, rectified

**Comment by wx-03** [comment]
lgtm

**Pull request merged**